### DISCUSSION TITLE
Normal that you have to rebuild Immich with every major update?

### DISCUSSION THREAD

**Discussion by mattish91** [immich-app/immich#13726] [Q&A]
Every few updates i need to set everything up from the begining (my host is running CasaOS) is that normal? I know this is an some what unstable application for the moment, but it looks so damn good i can't imagine my life withoutit. How ever, i have tried to restore the app from a full VM backup but it still doesn't want to start up normally after that, not even if i set it up with version number. It's just that an update is available and it won't restart the image. I might just go as far as use the experimental install.sh in order to not having to re upload everything each time there is an update available. 

How do you manage your updates for Immich and have you been running into issues after rebooting the docker image?

Im really curiouse to know as i started to look in to PhotoPrism, but really that's not a replacement in any way to be honest, i wanted to get away from my Nextcloud instance to run something lighter for just photos in particular, and Immich seams to do the job unless i need to reboot the host or app itself for some reason. I also feel that this reset thing that i currently have going for me 🤣 would be so much faster to restore the images for me if i could sync everything from my current "backup" from my desktop, i have this as i obviously can't 100% rely on Immich at the moment as it for some reason chrashes from time to time and i need to restore everything (like once a month).

**Comment by bo0tzz** [comment]
That doesn't sound normal at all. I don't know casaos, does it have some custom app store thing or are you just using the standard docker compose deployment?

**Reply by mattish91** [reply]
It does have a "app store" how over i have tried to use the compose directly with the same result, this probably is the 6th time i have to re install it with no other option but to actually reset everything.

Now i seam to not being able to set it up at all receiving an error from the db

```
immich-redis  | 1:M 24 Oct 2024 23:18:25.094 # Server initialized
immich-redis  | 1:M 24 Oct 2024 23:18:25.094 # WARNING Memory overcommit must be enabled! Without it, a background save or replication may fail under low memory condition. Being disabled, it can can also cause failures without low memory condition, see https://github.com/jemalloc/jemalloc/issues/1328. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
immich-redis  | 1:M 24 Oct 2024 23:18:25.094 * Ready to accept connections
immich-redis  | 1:signal-handler (1729812037) Received SIGTERM scheduling shutdown...
```

At some point i might just stop trying, it's getting really annoying.

**Reply by mmomjian** [reply]
This is not a problem, that's a normal message "WARNING Memory overcommit must be enabled". You'll need to give us more info/figure out what the actual problem is, or use a VM like you mentioned.

**Reply by mattish91** [reply]
seams like the database doesn't start, and as sson as it does start it just dies with the following explanation. (tried to start the postgresql service manually and then it takes like 5 minutes and it's dead again.

`# ./etc/init.d/postgresql start
No PostgreSQL clusters exist; see "man pg_createcluster" ... (warning).
`

I suppose this is a normal warning as well?

The last log i get in the log is this:
```
[33m[Api:Bootstrap][39m [32mImmich Server is listening on http://[::1]:2283 [v1.118.2] [production] 
[39m
```
It looks like it just cuts off when it's supposed to print something out in the log. really tricky to figure out whats actually going wrong, nd this time i copied the entire docker-compose.yml over to the server trough terminal using nano to create the server, it also looks like it's running but immich-postgres says "health: starting"

`8937d0ffda66   tensorchord/pgvecto-rs:pg14-v0.2.0               "docker-entrypoint.s…"   30 minutes ago   Up 7 minutes (health: starting)   5432/tcp
                                                                      immich-postgres`
Anyways, just got it setup on a LXC that ill try out for a few days just to make sure it's stable enough for me ^^ seams to be working at te moment at least.

**Comment by mmomjian** [comment]
I’ve been using since 1.8x something and I never had to re setup. You might be better off switching to a better OS like Debian. We hear a lot of Casa OS complaints on our discord.

**Reply by mattish91** [reply]
Might actually try to setup a completely blank VM for just Immich then.

**Comment by adambeck7** [comment]
Definitely a Casa issue. I've been running Immich for over 2 years via docker-compose on Unraid. Wonder if you don't have volume mounts set properly?

**Reply by mattish91** [reply]
As it does work until the upgrade im not sure what's causing this, how ever i now receive a new type of issue with the db even tho i have 9GB ram set for it.

**Reply by T0ytoy** [reply]
I use Immich on a mini-pc with 8 GB of RAM shared between multiple services, it works perfectly. Updates are going without a hitch using the official docker-compose instructions, special actions are only necessary when a breaking change is introduced (and documented!), of course.
